BrainPortal/public/javascripts/userfiles.js

```diff
@@ -401,6 +439,45 @@ $(function() {
 
     /* Show/Hide dynamic actions/menu elements according to current selection */
     (function () {
+      function launch_task() {
+        var parameters    = window.location.search.split(/\?|&/);
+        var launch_button = $(this).find('input.launch_tool');
+
+        var prepare_tool_id = undefined
+        for (var i = 0; i < parameters.length; i++) {
+          var [name, id]     = parameters[i].split(/=/);
+          var userfile_checkboxes = $("input[name='file_ids[]']");
+
+          if (name != "prepare_tool_id") {
+            continue;
+          }
+
+          prepare_tool_id = id;
```

Comment:
Once you found the tool_id, break out of the loop instead of continuing to scan all the other params